ratis-examples/src/main/java/org/apache/ratis/examples/counter/server/CounterStateMachine.java

```diff
@@ -139,23 +143,28 @@ public long takeSnapshot() {
           + "\", last applied index=" + state.getApplied());
     }
 
+    // update storage
+    final MD5Hash md5 = MD5FileUtil.computeAndSaveMd5ForFile(snapshotFile);
```

Comment:
Is there any reader of the saved md5 file? Does the file have an owner that remove it on exiting?

Reply:
> should be removed

Can you show me the code path to implement this clean-up logic? I don't find it :(

Reply:
Currently, we don't have clean-up logic implemented in the examples.

Reply:
@tisonkun , actually, the old snapshots are deleted in `StateMachineUpdater.takeSnapshot()` according to the `SnapshotRetentionPolicy`.  It should also delete the md5 files.

Reply:
Yes. I saw aabe571c2915d32cc9a479546477bde0411315f5 now :)